mooncake-store/include/client.h

```diff
@@ -364,4 +364,34 @@ class Client {
     UUID client_id_;
 };
 
+/**
+ * @brief Fluent builder for configuring a mooncake::Client instance.
+ *
+ * Provides readable, type-safe setters with sensible defaults so callers can
+ * specify only the options they need while reusing existing Client::Create
+ * logic under the hood.
+ */
+class MooncakeStoreBuilder {
```

Comment:
I think `ClientBuilder` should be a better name in the context, as we are infact building a `Client`, not a `MooncakeStore`.

Reply:
How about MooncakeStoreClientBuilder? Don't be afraid of a long name.